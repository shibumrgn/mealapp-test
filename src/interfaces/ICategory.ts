export interface ICategory {
    strCategory: string;
}
