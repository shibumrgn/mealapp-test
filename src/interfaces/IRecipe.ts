export interface IRecipe {
    idMeal: string;
    strMeal: string;
    strDrinkAlternate: string;
    strCategory: string;
    strArea: string;
    strInstructions: string;
    strMealThumb: string;
    strTags: string;
    strYoutube: string;
    ingredients: {
        name: string,
        amount: string
    }[]
}