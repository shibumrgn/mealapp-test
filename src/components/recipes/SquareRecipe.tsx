import {Dimensions, Image, StyleSheet, Text, View} from 'react-native';
import React from 'react';
import {IRecipe} from '../../interfaces/IRecipe';
import {Brand} from '../../constants/Brand';
const width = Dimensions.get('window').width / 2.5;
interface IProps {
    recipe: IRecipe;
}

const SquareRecipe = (props: IProps) => {
    return (
        <View style={styles?.parent}>
            <View
                style={styles.titleContainer}>
                <Text style={styles.title} numberOfLines={2}>
                    {props?.recipe?.strMeal}
                </Text>
                <View>
                    <Text>Time</Text>
                    <Text>10 Mins</Text>
                </View>
            </View>
            <Image
                source={{
                    uri: props.recipe?.strMealThumb,
                }}
                style={styles.image}
            />
        </View>
    );
};

export default SquareRecipe;

const styles = StyleSheet.create({
    parent: {
        width: width,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        padding: 5,
    },
    image: {
        width: width / 1.25,
        height: width / 1.25,
        borderRadius: width,
        borderColor: '#e5e5e5',
        borderWidth: 10,
        position: 'absolute',
    },
    title: {
        ...(Brand.tinyBold as Object),
        textAlign: 'center',
        marginBottom: 20,
        height: 40,
    },
    titleContainer: {
        backgroundColor: '#c4edec',
        width: '100%',
        borderRadius: 15,
        marginTop: width / 2,
        paddingTop: width / 3,
        paddingHorizontal: 10,
        paddingBottom: 10,
    }
});
