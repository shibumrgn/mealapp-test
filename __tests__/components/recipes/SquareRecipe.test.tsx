
import 'react-native';
import React from 'react'; 
import renderer from 'react-test-renderer';
import SquareRecipe from '../../../src/components/recipes/SquareRecipe';
import SampleRecipes from '../../SampleRecipes.json'
import { IRecipe } from '../../../src/interfaces/IRecipe';
 
 it('renders correctly', () => {
   const render = renderer.create(<SquareRecipe recipe={SampleRecipes[0] as any as IRecipe}/>)?.toJSON();
   expect(render).toMatchSnapshot();
 });
 