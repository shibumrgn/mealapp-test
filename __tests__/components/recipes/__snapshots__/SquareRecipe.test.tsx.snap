// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "width": 300,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://www.themealdb.com/images/media/meals/ustsqw1468250014.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 290,
        "height": 280,
        "width": 280,
      }
    }
  />
  <Text
    numberOfLines={2}
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Spicy Arrabiata Penne
  </Text>
</View>
`;
